compiler/resolution/functionResolution.cpp

```diff
@@ -7816,18 +7846,25 @@ static void insertReturnTemps() {
               tmp->addFlag(FLAG_INSERT_AUTO_DESTROY);
 
             contextCallOrCall->insertBefore(def);
+            def->insertAfter(new CallExpr(PRIM_MOVE,
+                                          tmp,
+                                          contextCallOrCall->remove()));
 
             if (fMinimalModules == false) {
+              if (isIteratorOrForwarder(fn)) {
+                handleStatementLevelIteratorCall(def, tmp);
+
+              } else
               if ((fn->retType->getValType() &&
                    (isSyncType(fn->retType->getValType()) ||
                     isSingleType(fn->retType->getValType())))         ||
 
                   isSyncType(fn->retType)                             ||
-                  isSingleType(fn->retType)                           ||
-                  fn->isIterator()) {
+                  isSingleType(fn->retType)                           )
+              {
                 CallExpr* sls = new CallExpr("_statementLevelSymbol", tmp);
 
-                contextCallOrCall->insertBefore(sls);
+                def->next->insertAfter(sls);
```

Comment:
I don't see the point of the change on this line and like the previous version better. Am I missing something?